src/common/tse.c

```diff
@@ -803,9 +797,6 @@ tse_task_complete(tse_task_t *task, int ret)
 	/** update task in scheduler lists. */
 	if (!dsp->dsp_cancelling && done)
 		tse_sched_process_complete(dsp);
-	/** If another thread canceled, make sure we drop the ref count */
-	else if (bumped)
-		tse_sched_decref(dsp);
 
 	/** -1 from tse_task_create() if it has not been reinitialized */
 	if (done)
```

Comment:
I think here is the issue:
if the task reinitialized itself in completion callback, then the task is going to be scheduled again, which will take refcount:
```
tse_task_schedule(tse_task_t *task, bool instant)
...
        } else {
                /** Otherwise, scheduler will process it from init list */
                d_list_add_tail(&dtp->dtp_list, &dsp->dsp_init_list);
        }
        tse_sched_addref_locked(dsp);
...
```
and the refcount is only decreased for once no matter how many times the task has been rescheduled.

In addition, some comments are out-of-date, we definitely want to update them.

Reply:
by check some code, the task reinit need not do the tse_task_schedule. tse_task_schedule should only be called once to insert one task to scheduler.
If user calls tse_task_schedule() multiple times for same task, it possibly cause some problems, as now there is no special logic to handle that case. Do we need to add some code for that case? Or it can be treated as programming error and user take the responsibility? Either is fine to me, for simplicity maybe fine to leave it as now.